ci: update cilium-cli to v0.10.4

tklauser opened the pull request:
See individual commits.

tklauser committed:
ci: update cilium-cli to v0.10.4

Signed-off-by: Tobias Klauser <tobias@cilium.io>

pchaigno commented on .github/workflows/conformance-externalworkloads-v1.10.yml:
```diff
@@ -65,7 +65,7 @@ env:
   zone: us-west2-a
   vmStartupScript: .github/gcp-vm-startup.sh
-  cilium_cli_version: v0.10.3
+  cilium_cli_version: v0.10.4
```
Don't we need to set the base version here as well?

tklauser replied:
Good catch, I missed this one. Changed to set base version v1.10.

pchaigno commented on .github/workflows/conformance-externalworkloads-v1.11.yml:
```diff
@@ -65,7 +65,7 @@ env:
   zone: us-west2-a
   vmStartupScript: .github/gcp-vm-startup.sh
-  cilium_cli_version: v0.10.3
+  cilium_cli_version: v0.10.4
```
Same here.

tklauser replied:
Changed to set base version v1.11.

pchaigno commented on .github/workflows/conformance-multicluster-v1.10.yaml:
```diff
@@ -147,9 +147,12 @@ jobs:
           CLUSTERMESH_ENABLE_DEFAULTS="--apiserver-image=quay.io/${{ github.repository_owner }}/clustermesh-apiserver-ci \
             --apiserver-version=${SHA}"
+          CONNECTIVITY_TEST_DEFAULTS="--base-version=v1.10 \
+            --flow-validation=disabled"
```
Why are we disabling flow validation here?

tklauser replied:
Mindless copy&paste, it seems. Reverted that change.

tklauser committed:
ci: set base Cilium version in connectivity tests on v1.10 and v1.11

Starting with cilium-cli v0.10.4, `cilium connectivity test` needs to be
aware of the Cilium version it runs tests against. Normally, the version
can be derived from the image tag, but for CI images the version is not
a semver but just a hash. In these cases, the base Cilium version needs
to be specified using the hidden `--base-version` flag, akin to the
`cilium install` flag already used in CI.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
